part_2/js/script.js

```diff
@@ -0,0 +1,201 @@
+                let elementP2 = document.createElement("p"); //create p element
+
+                elementH1.innerHTML = `${employee.first_name} ${employee.last_name}`; //h1 element with first and last name
+                elementH2.innerHTML = `Department: ${employee.department.name}`; //h2 element department name
+                elementP1.innerHTML = `Annual Salary: ${employee.annual_salary}`; //p1 element annual salary
+                elementP2.innerHTML = `Hire Date: ${employee.hire_date}`;//p2 element hire date
+
+                clearContainer(right); //clear right container before adding data
+                right.appendChild(elementH1);
+                right.appendChild(elementH2);
+                right.appendChild(elementP1);
+                right.appendChild(elementP2);
+
+                //Begin Step 3
+                fetchIMG(employee.image_filename);//call function fetchIMG
+                //End Step 3
+
+                //Begin Step 4
+                addURL();//call function addAnchor
+                let deptURL = document.getElementById("deptAnchor"); //deptURL variable to id deptAnchor
+                deptURL.addEventListener("click", ()=> {//event listener when lick deptAnchor id
+                    getDepartmentList(employee); //invoke function getDepartmentList
+                });
+                //End Step 4
+            })
+                .catch(error => {
+                    console.log("Error during fetch: " + error); //log o console error
+                });
+
+        }else {//if value is blank
+            clearContainer(right); //clear right container
+        }
+    }//end of onChange Function
+    //End Step 2
+
+    //Begin Step 3
+    function fetchIMG(imgURL){//function to fetch image
+        fetch(imgURL, {
+            method: "GET", //recieve data with GET
+            mode: "cors", //allow outside domain
+            redirect: "follow" //follow the redirect
+        })
+        .then((response) => {//response received
+            if(response.ok){
+                console.log(response); //log the response to console
+                return response.blob(); //return as a blob
+            }
+            throw new Error(response.statusText); //throw error of response
+        })
+        .then((blob) => {
+            let elementIMG = new Image(); //create img element
+            elementIMG.src = URL.createObjectURL(blob); //set src of img element to blob
+            right.appendChild(elementIMG); //append img to right div
+        })
+        .catch(error => console.log("Error when fetching image: " + error));
+    }
+    //End Step 3
+
+    //Begin Step 4
+    function addURL(){//function addURL to add url to doc
+        let deptURL = document.createElement("a"); //create url element
+        let tagIMG = document.getElementsByTagName("img")[0]; //img by tag name
+        deptURL.innerText = "View Department"; //inner text of url element
+        deptURL.id = "deptURL"; //set id 
```

Comment:
deptURL is going to need an href ("#" at least) to display as a link - and a listener in order for a function - your getDepartmentList() to be called. You call that function onClick of the element with the ID deptAnchor. If you want that to happen for this input - you'd need to assign this anchor tag that ID.